script/pipelines/pipeline-template.yml

```diff
@@ -150,3 +162,62 @@ stages:
               - publish: output
                 displayName: "Publish packages"
                 artifact: "packages"
+
+      - stage: Stage
+        dependsOn:
+          - Package
+          - Validate
+        jobs:
+          - job: publish_go
+            displayName: "Publish Go Wrapper"
+            pool: ${{ parameters.LinuxPool }}
+            steps:
+              - checkout: self
+                fetchTags: true
+                persistCredentials: true
+              - download: current
+                artifact: packages
+                patterns: "**"
+              - bash: |
+                  version=cargo metadata --format-version 1 | jq -r '.packages[] | select(.name == "cosmoscx") | .version'
+                  tag_name="go/azcosmoscx/v$version"
+                  branch_name="azp-release-staging/v$version"
+                  echo "##vso[task.setvariable variable=ReleaseVersion]$version"
+                  echo "##vso[task.setvariable variable=ReleaseTag]$tag_name"
+                  echo "##vso[task.setvariable variable=ReleaseBranch]$branch_name"
+              - bash: |
+                  echo "Preparing Go wrapper release tag '$(ReleaseTag)'"
+                  if git rev-parse -q --verify "refs/tags/$(ReleaseTag)" >/dev/null; then
+                    echo "Error: Git tag '$(ReleaseTag)' already exists. Aborting release."
+                    exit 1
+                  fi
+
+                  git checkout -b "$(ReleaseBranch)"
+
+                  src_dir="$(Pipeline.Workspace)/packages/vendored"
+                  dest_dir="./go/azcosmoscx/libcosmoscx-vendor"
+
+                  echo "Creating destination directory '$dest_dir'"
+                  mkdir -p "$dest_dir"
+
+                  # Ensure there is at least one subdirectory to copy
+                  if ! find "$src_dir" -maxdepth 1 -mindepth 1 -type d -print -quit >/dev/null 2>&1; then
+                    echo "No subdirectories found in '$src_dir' to copy."
+                    exit 1
+                  fi
+
+                  echo "Copying subdirectories from '$src_dir' into '$dest_dir' (subdirs become direct children)"
+                  # rsync copies the contents of src_dir into dest_dir, making each top-level subdirectory a direct child
+                  rsync -a --chmod=ugo=rwX "$src_dir"/ "$dest_dir"/
+
+                  echo "Copy complete."
+                displayName: "Place vendored binaries"
+              - bash: |
+                  git add go/azcosmoscx/libcosmoscx-vendor
+                  git config --global user.name "Azure Pipelines"
+                  git config --global user.email "bot@example.com"
```

Comment:
This seems a little sketch.

Reply:
Yeah, I agree. I'll see what I can work out for this